spec/app_support/order_row_importer_spec.rb

```diff
@@ -0,0 +1,271 @@
+require "spec_helper"
+
+describe OrderRowImporter do
+  subject { OrderRowImporter.new(row, order_import) }
+  let(:account) do
+    create(:nufs_account,
+      account_users_attributes: [ attributes_for(:account_user, user: user) ],
+    )
+  end
+  let(:facility) { create(:facility) }
+  let(:facility_account) do
+    facility.facility_accounts.create(attributes_for(:facility_account))
+  end
+  let(:order_import) { build(:order_import, creator: user, facility: facility) }
+  let(:service) do
+    create(:setup_service,
+      facility: facility,
+      facility_account: facility_account,
+    )
+  end
+  let(:user) { create(:user) }
+
+  let(:row) {{
+    "Netid / Email" => username,
+    "Chart String" => chart_string,
+    "Product Name" => product_name,
+    "Quantity" => quantity,
+    "Order Date" => order_date,
+    "Fulfillment Date" => fulfillment_date,
+    "Errors" => errors,
+    "Note" => notes,
+  }}
+
+  let(:username) { "column1" }
+  let(:chart_string) { "column2" }
+  let(:product_name) { "column3" }
+  let(:quantity) { "column4" }
+  let(:order_date) { "column5" }
+  let(:fulfillment_date) { "column6" }
+  let(:errors) { "column7" }
+  let(:notes) { "column8" }
+
+  describe "#import" do
+    shared_examples_for "an order was created" do
+      it "creates an order" do
+        expect { subject.import }.to change(Order, :count).by(1)
+      end
+    end
+
+    shared_examples_for "an order was not created" do
+      it "does not create an order" do
+        expect { subject.import }.not_to change(Order, :count)
+      end
+
+      it "does not add an order_detail" do
+        expect { subject.import }.not_to change(OrderDetail, :count)
+      end
+    end
+
+    context "when the fulfillment date is invalid" do
```

Comment:
Lately, I've been liking to explicitly declare what's invalid within the context. That way if somebody else changes something at a higher level, this test doesn't get affected. Also, it makes traveling up the stack to find why it's invalid easier.